test/rollup_circuit.js

```diff
@@ -453,65 +608,1056 @@ describe("Rollup Basic circuit TXs", function () {
+            ethAddress: account2.ethAddress,
+            onChain: true
+        });
+        
+        await bb.build();
+        const input = bb.getInput();
+        
+        const w = circuit.calculateWitness(input, {logTrigger:false, logOutput: false, logSet: false});
+        checkBatch(circuit, w, bb);
+        
+        await rollupDB.consolidate(bb);
+        
+        const bb2 = await rollupDB.buildBatch(NTX, NLEVELS);
+        
+        const tx = {
+            fromIdx: 1,
+            toIdx: 2,
+            loadAmount: 100,
+            coin: 0,
+            amount: 50,
+            nonce: 0,
+            userFee: 10
+        };
+        account1.signTx(tx);
+        bb2.addTx(tx);
+        bb2.addCoin(0, 5);
+        try {
+            await bb2.build();
+            const input2 = bb2.getInput();
+            const w2 = circuit.calculateWitness(input2, {logTrigger:false, logOutput: false, logSet: false});
+            checkBatch(circuit, w2, bb2);
+        } catch (error) {
+            assert.include(error.message, "Load ammount must be 0 for offChainTxs");
+        }
+
+    });
+    it("Should check onchain transfer", async () => {
+        // Start a new state
+        const db = new SMTMemDB();
+        const rollupDB = await RollupDB(db);
+        const bb = await rollupDB.buildBatch(NTX, NLEVELS);
+                
+        const account1 = new RollupAccount(1);
+        const account2 = new RollupAccount(2);
+                
+        bb.addTx({
+            fromIdx: 1,
+            loadAmount: 1000,
+            coin: 0,
+            ax: account1.ax,
+            ay: account1.ay,
+            ethAddress: account1.ethAddress,
+            onChain: true
+        });
+                
+        bb.addTx({
+            fromIdx: 2,
+            loadAmount: 2000,
+            coin: 0,
+            ax: account2.ax,
+            ay: account2.ay,
+            ethAddress: account2.ethAddress,
+            onChain: true
+        });
+                
+        await bb.build();
+        const input = bb.getInput();
+
+        const w = circuit.calculateWitness(input, {logTrigger:false, logOutput: false, logSet: false});
+        checkBatch(circuit, w, bb);
+                
+        await rollupDB.consolidate(bb);
+                
+        const bb2 = await rollupDB.buildBatch(NTX, NLEVELS);
+                
+        const tx = {
+            fromIdx: 2,
+            toIdx: 1,
+            coin: 0,
+            amount: 50,
+            nonce: 0,
+            ax: account2.ax,
+            ay: account2.ay,
+            ethAddress: account2.ethAddress,
+            onChain: true
+        };
+        bb2.addTx(tx);
+
+        await bb2.build();
+        const input2 = bb2.getInput();
+        const w2 = circuit.calculateWitness(input2, {logTrigger:false, logOutput: false, logSet: false});
+        checkBatch(circuit, w2, bb2);
+
+    });
+
+    it("Should check onchain transfer to 0 with the wrong account", async () => {
```

Comment:
If smart contracts does not allow this type of transaction, maybe it should not be considered in our test

Reply:
deleted too^^